### DISCUSSION TITLE
Scrollview inside of Stackpanel doesn't work.

### DISCUSSION THREAD

**Discussion by weltraumreise** [AvaloniaUI/Avalonia#6350] [Q&A] [answered]
I have such an axaml in my page. The ScrollViewer doesn't work here. The scrollbar will not show up if I expanded the TreeView structure beyond the view. But it will work, if I replaced the StackPanel with Panel, DockPanel or Grid.
Am I missing anything to make it work?
```
    <DockPanel>
        <Border>
            <StackPanel Orientation="Vertical">
                <ScrollViewer  DockPanel.Dock="Left">
                    <TreeView Items="{Binding Root.Children}">
                        <TreeView.ItemTemplate>
                            <TreeDataTemplate ItemsSource="{Binding Children}" DataType="models:TreeNode">
                                <TextBlock Text="{Binding Value }" />
                            </TreeDataTemplate>
                        </TreeView.ItemTemplate>
                    </TreeView>
                </ScrollViewer>
            </StackPanel>
        </Border>
    </DockPanel>

**Comment by jp2masa** [comment, marked as answer]
That's because `StackPanel` measures with infinite space in the `Orientation` direction, so the `ScrollViewer` will always expand indefinitely and the content always fits inside the available space. In this case you would probably want to place the `StackPanel` inside the `ScrollViewer`. Also, `DockPanel.Dock` won't do anything in this case, you probably want to place it on the `Border`. So something like this:

```xaml
        <Border DockPanel.Dock="Left">
            <ScrollViewer>
                <StackPanel Orientation="Vertical">
                    ...
                </StackPanel>
            </ScrollViewer>
        </Border>
```

**Reply by weltraumreise** [reply]
Thank you very much. It is a clear explanation.
